### ISSUE TITLE
gofabric8 deploy -y --domain=foo.bar should make sure it updates the fabric8 oauthclient if one exists with a different domain?

### ISSUE THREAD

**Issue by jstrachan** [fabric8io/gofabric8#17]
Or maybe just delete it and force it to be recreated?

**Comment by mrosic** [comment]
Yes, please fix that.

People forget to do an "oc delete oauthclient fabric8" because "oc get all" doesn't show oauthclients and OpenShift documentation is far from perferct.

The current situation is that if you ever do a "gofabric8 deploy" with a wrong --domain or without --domain then you'll never get fabric8 to work... unless you do a "oc delete oauthclient fabric8".

**Comment by hodrigohamalho** [comment]
I think i'm facing this issue.

I have a native installation of openshift V3.0.2, and trying to install fabric8.
But, when I try to open fabric on my web browser it redirects me to login page even if I currently logged in.

Look at the oauthclients:
http://postimg.org/image/z7e1birw1/full/

Running gofabric8 validate I get an error too:
http://postimg.org/image/s64kw94rv/full

Any hint?

**Comment by jstrachan** [comment]
@hodrigohamalho the serviceaccount looks fine to me - if your domain is `cloud.devops.org`; though the second output shows master.devops.org so maybe you wanted the domain to be just `devops.org` instead of `cloud.devops.org`?

If you want to switch domains then try deleting the fabric8 ServiceAccount and rerunning gofabric8
```
oc delete sa fabric8
gofabric8 deploy -y
gofabric8 secrets -y
oc get sa
```

**Comment by jstrachan** [comment]
BTW a work around for this issue for now is to type:
```
oc delete sa fabric8
```
to delete the ServiceAccount prior to running `gofabric8 deploy` with the different domain

**Comment by hodrigohamalho** [comment]
@jstrachan thanks for you attention.

The problem was CORS. 
I added the entry below on ```/etc/openshift/master/master-config.yaml``` and worked. I think it should be more explicit on documentation.
```
corsAllowedOrigins:
  - .*
```

**Comment by jstrachan** [comment]
Agreed. To be fair the docs do mention passing cors-allowed: http://fabric8.io/guide/getStarted/openshift.html#alternative-installation-route

I've added a trouble shooting section to the docs though to make this more clear:
https://github.com/jstrachan/fabric8/blob/fixes/docs/getStarted/openshift.md#troubleshooting

**Comment by jstrachan** [comment]
I've just fixed this - its available in the new 0.4.21 release. So now `gofabric8 deploy` will delete the fabric8 OAuthClient first to ensure that after the fabric8 console is installed its got the latest greatest domain etc
